### PR TITLE
Publish SOTS gridded multiyear data in new sub-directory

### PR DESCRIPTION
**Pull request by leonardolaiolo**
@mhidas 
I could not use `_get_nc_att` because it was failing some unittest. Indeed, it was raising an `InvalidFileContentError` (you can have a look at my first attempt in commit 435d59c) for the files that do not have 'data_product_type' as attribute. I then avoid using '_get_nc_att' and used instead `_open_nc_file` and `hasattr` to avoid the `InvalidFileContentError` (commit ade1b3f)

### CODE REVIEW

File: aodndata/moorings/classifiers.py [hunk 1]
```diff
@@ -324,9 +326,13 @@ def dest_path(cls, input_file):
             dir_list += ['ASFS', 'SOFS', cat, 'Real-time', rt_folder_name]
         elif subfac in ('SOTS', 'ASFS'):
             dir_list.append('SOTS')
-            dir_list.append(cls._get_deployment_year(input_file))
-            if cls._is_realtime(input_file):
-                dir_list.append('real-time')
+            # Check for data_product_type attribute
+            if cls._get_nc_att(input_file, 'data_product_type'):
```
Comment (mhidas) [inline aodndata/moorings/classifiers.py]: As it is, `cls._get_nc_att` will raise an error if the attribute is missing. What you want here is for it to return `False` in that case, which you can achieve like this:
```python
         if cls._get_nc_att(input_file, 'data_product_type', default=False):
```
See the docstring for `_get_nc_att`: https://github.com/aodn/python-aodncore/blob/d78192d877e3a07d7a8df178cc45ed502a22bb8f/aodncore/pipeline/fileclassifier.py#L75

Comment (mhidas) [reply aodndata/moorings/classifiers.py]: You will also need to save and check the value of the attribute -- see https://github.com/aodn/PO-Backlog/issues/3620#issuecomment-2164241924

**Review by mhidas — CHANGES_REQUESTED** [review]
Thanks @leonardolaiolo . Getting there, but as you will see, we had the requirements slightly wrong (sorry...)

**Review by mhidas — APPROVED** [review]

**Comment by mhidas** [comment]
Remember in order to test this in RC you'll need to promote the newly built version of the aodndata pacgate at https://build.aodn.org.au/job/python-aodndata_artifact_promote/

**Comment by leonardolaiolo** [comment]
Good point!

**Pull request merged**